docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -287,3 +287,6 @@ An unused bean:
 Users can instruct the container to not remove any of their specific beans (even if they satisfy all the rules specified above) by annotating them with `io.quarkus.arc.Unremovable`.
  
 Furthermore, extensions can eliminate possible false positives by producing `UnremovableBeanBuildItem`.
+
+Finally, Quarkus provides middle ground for this optimization where all application beans are kept whether or not they are unused, but the optimization
+proceeds normally for non application classes. To use this mode set: `quarkus.arc.remove-unused-beans=fwk`.
```

Comment:
I don't think we need a full name... maybe we could support few more aliases such as `fwk` and `framework`?

Reply:
why would we use a cryptic thing instead of `framework`? It's not as if it was long to type and at least you understand what it is without thinking twice about it.

Reply:
I agree we can have both. I will go ahead and add that :)

Reply:
Added `framework` as well as `fwk`